crates/primitives/transactions/src/into_starknet_api.rs

```diff
@@ -294,33 +535,70 @@ fn version(version: &Felt) -> starknet_api::transaction::TransactionVersion {
-        }
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::tests::{
+        dummy_l1_handler, dummy_tx_declare_v0, dummy_tx_declare_v1, dummy_tx_declare_v2, dummy_tx_declare_v3,
+        dummy_tx_deploy, dummy_tx_deploy_account_v1, dummy_tx_deploy_account_v3, dummy_tx_invoke_v0,
+        dummy_tx_invoke_v1, dummy_tx_invoke_v3,
+    };
+
+    #[test]
+    fn test_into_api_transaction() {
+        let tx: Transaction = dummy_tx_invoke_v0().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_tx_invoke_v1().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_tx_invoke_v3().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_l1_handler().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_tx_declare_v0().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_tx_declare_v1().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_tx_declare_v2().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_tx_declare_v3().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_tx_deploy().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_tx_deploy_account_v1().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
+
+        let tx: Transaction = dummy_tx_deploy_account_v3().into();
+        let api_tx = starknet_api::transaction::Transaction::try_from(&tx).unwrap();
+        let tx_back: Transaction = api_tx.into();
+        assert_eq!(tx, tx_back);
```

Comment:
would love an util function that could be used accross *ALL* tests that convert then convert back to ensure no data is lost during marshalling. 
Something like `assert_consistent_conversion<T1, T2>(a, b)` wdyt ? Would help a lot with clarity

Reply:
like that?
```rust
pub fn assert_consistent_conversion<T1, T2>(a: T1)
where
	T1: Clone + PartialEq + std::fmt::Debug + From<T2>,
	T2: Clone + PartialEq + std::fmt::Debug + From<T1>,
{
	let b: T2 = a.clone().into();
	let c: T1 = b.clone().into();
	assert_eq!(a, c);
	let d: T2 = c.clone().into();
	assert_eq!(b, d);
}
```

Reply:
yup should be good

Reply:
just change on the other places where it can be used